server/base/src/main/java/org/apache/accumulo/server/master/state/MetaDataStateStore.java

```diff
@@ -139,6 +140,11 @@ public void suspend(Collection<TabletLocationState> tablets,
       for (TabletLocationState tls : tablets) {
         Mutation m = new Mutation(tls.extent.getMetadataEntry());
         if (tls.current != null) {
+          // if the location more is assignment, then preserve the current location in the last
+          // location value
+          if ("assignment".equals(context.getConfiguration().get(Property.TABLE_LOCATION_MODE))) {
+            tls.current.putLastLocation(m);
+          }
```

Comment:
This is definitely reading it from the system configuration, not the per-table configuration, so it should be a `general.` property, not a `table.` one.

Reply:
oh, did not realize that.  ok.